### PR TITLE
Commonized IR serializer infrastructure between native and js

### PR DESCRIPTION
**Pull request by alexander-gorshenev**

### CODE REVIEW

File: compiler/ir/backend.common/src/KotlinIr.proto [hunk 1]
```diff
@@ -74,6 +74,11 @@ message IrDeclarationOrigin {
   }
 }
 
+message Name {
```
Comment (udalov) [inline compiler/ir/backend.common/src/KotlinIr.proto]: Why isn't it enough to encode names as strings with `Name.guessByFirstCharacter` as a way to distinguish special names from identifiers?

Comment (alexander-gorshenev) [reply compiler/ir/backend.common/src/KotlinIr.proto]: Should be enough, I suppose. Thanks.

**Comment by alexander-gorshenev** [comment]
For reference, the corresponding native PR is here: https://github.com/JetBrains/kotlin-native/pull/2769

File: compiler/ir/backend.common/src/KotlinIr.proto (hunk 1 above)
Comment (alexander-gorshenev) [reply compiler/ir/backend.common/src/KotlinIr.proto]: Moved to use Name.guessByFirstCharacter().

**Comment by udalov** [comment]
Has this been reviewed by someone else?

**Comment by romanart** [comment]
@udalov Yes, in [upsource](https://upsource.jetbrains.com/kotlin/review/KOTLIN-CR-2822)

**Pull request merged**